x-pack/test/plugin_api_integration/test_suites/task_manager/check_registered_task_types.ts

```diff
@@ -110,6 +110,7 @@ export default function ({ getService }: FtrProviderContext) {
         'cleanup_failed_action_executions',
         'cloud_security_posture-stats_task',
         'dashboard_telemetry',
+        'endpoint:check-deleted-files-task',
```

Comment:
Did not know we needed to do this 👍

Reply:
> Reviewed and left some comments - the primary issue I see is that you are not iterating over the file chunks when you query them (each call to ES I think only bring back 10 hits at a time).

Very good point. Will adjust.

> The other concern here is with perhaps too much refactoring if we commit this now as is and then move it to Fleet. Several references will have to be redone and equivalents found on the fleet side.

yeah, this makes sense. I'm also thinking I can have fleet version of it PR'ed by tomorrow EOD so it is probably best to just drop this one. I'll leave it open for now in case anyone else has comments on the logic but will just close this out when I open the fleet one hopefully tomorrow.

Thanks for the review 🙆‍♂️.